tests/checker/test_https.py

```diff
@@ -15,34 +15,59 @@
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 """
-Test news checking.
+Test https.
 """
-import pytest
+from OpenSSL import crypto
 
 from tests import need_network
-from . import LinkCheckTest
+from .httpserver import HttpsServerTest, CookieRedirectHttpRequestHandler
+from .. import get_file
 
+from linkcheck import httputil
 
-class TestHttps (LinkCheckTest):
+
+class TestHttps(HttpsServerTest):
     """
     Test https: link checking.
     """
 
-    @pytest.mark.xfail
+    def __init__(self, methodName='runTest'):
+        super(TestHttps, self).__init__(methodName=methodName)
+        self.handler = CookieRedirectHttpRequestHandler
```

Comment:
Please don't overwrite `__init__`.

You can assign `handler = CookieRediectHttpRequestHandler` as a class attribute.

Reply:
I just copied tests/checker/test_http.py. test_http_redirect.py and test_https_redirect.py also overwrite `__init__`. 
The parent class, HttpsServerTest, does use `self.handler`, with the `__init__` being in its parent, HttpServerTest. So, there is a fair amount to rewrite here that is possibly best done in a new PR. Would you have a class attribute in the parent and then override it in the children?

Reply:
Ah, ok then. Consistency with existing code trumps general guidelines.